pkg/standalone-utils/contracts/PauseHelper.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import { IVault } from "@balancer-labs/v3-interfaces/contracts/vault/IVault.sol";
+import { IVaultAdmin } from "@balancer-labs/v3-interfaces/contracts/vault/IVaultAdmin.sol";
+import { SingletonAuthentication } from "@balancer-labs/v3-vault/contracts/SingletonAuthentication.sol";
+
+contract PauseHelper is SingletonAuthentication {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    event PoolAdded(address pool);
+    event PoolRemoved(address pool);
+
+    EnumerableSet.AddressSet private _poolsSet;
+
+    constructor(IVault vault) SingletonAuthentication(vault) {}
+
+    // --------------------------  Manage Pools --------------------------
+
+    /**
+     * @notice Add pools to the list of pools that can be paused
+     * @param newPools List of pools to add
+     */
+    function addPools(address[] calldata newPools) external authenticate {
+        uint256 length = newPools.length;
+
+        for (uint256 i = 0; i < length; i++) {
+            _poolsSet.add(newPools[i]);
+
+            emit PoolAdded(newPools[i]);
+        }
+    }
+
+    /**
+     * @notice Remove pools from the list of pools that can be paused
+     * @param pools List of pools to remove
+     */
+    function removePools(address[] memory pools) public authenticate {
+        uint256 length = pools.length;
+        for (uint256 i = 0; i < length; i++) {
+            _poolsSet.remove(pools[i]);
+
+            emit PoolRemoved(pools[i]);
+        }
+    }
+
+    /**
+     * @notice Pause pools
+     * @param pools List of pools to pause
+     */
+    function pause(address[] memory pools) public authenticate {
+        uint256 length = pools.length;
+        for (uint256 i = 0; i < length; i++) {
+            require(_poolsSet.contains(pools[i]), "Pool is not in the list of pools");
+
+            getVault().pausePool(pools[i]);
+        }
+    }
+
+    // --------------------------  Getters --------------------------
+    /**
+     * @notice Get the number of pools
+     * @return Number of pools
```

Comment:
```suggestion
     * @notice Get the number of pools.
     * @dev Needed to support pagination in case the list is too long to process in a single transaction.
     * @return poolCount The current number of pools in the pausable list
```